csp-mixin/signals/loadbalancer.libsonnet

```diff
@@ -0,0 +1,61 @@
+local commonlib = import 'common-lib/common/main.libsonnet';
+function(this)
+  {
+    local s = self,
+    filteringSelector: this.filteringSelector,
+    groupLabels: this.groupLabels,
+    instanceLabels: this.instanceLabels,
```

Comment:
This is where you can override the labels which are used. Take a look at [this documentation](https://github.com/grafana/jsonnet-libs/tree/master/common-lib/common/signal#configuration-options) for details, and I'm happy to answer any questions that come up when you start working with it.

Reply:
@rgeyer do you know how can I define a different list of `groupLabels` depending of the metric being used? For example the following metrics not contains the **client_country** label:

- stackdriver_https_lb_rule_loadbalancing_googleapis_com_https_backend_response_bytes_count
- stackdriver_https_lb_rule_loadbalancing_googleapis_com_https_backend_request_bytes_count

I would like to define a different array depending of the metric but no idea how to access that value.

Reply:
@yduartep , in signal itself you can add `aggKeepLabels` to keep extra labels for specific signals. This affects aggregation if you use `%(agg)s`  template for example or type != raw. See docs above for details.

In general, If you want different groupLabels then you can split signals file into two and define groupLabels separately.

Reply:
Hi @v-zhuravlev and @rgeyer I could define multiple signals but then I want to have just one dashboard composed of multiple signals because the filters are different on a couple of panels defined on the same dashboard.

Do you know if in the `main.libsonnet` file I can define my dashboard concatenating or merging multiple signals like:

```
 loadbalancer: commonlib.signals.unmarshallJsonMulti(this.config.signals.loadbalancer, type=this.config.metricsSource) + commonlib.signals.unmarshallJsonMulti(this.config.signals.loadbalancerOther, type=this.config.metricsSource),
```

Currently, I am defining the filters manually just for the panel that have different list of filters and is working like on the next example. I don't know if is a good practice:

```
errorRate: {
        name: 'Error rate visualization',
        description: 'Error rate visualization',
        type: 'raw',
        sources: {
          stackdriver: {
            expr: 'sum(stackdriver_https_lb_rule_loadbalancing_googleapis_com_https_request_count{job="integrations/gcp",job=~"$job",project_id=~"$project_id",client_country=~"$client_country",backend_target_name=~"$backend_target_name", response_code_class!="200", response_code_class!="0"}) / sum(stackdriver_https_lb_rule_loadbalancing_googleapis_com_https_request_count{%(queriesSelector)s})',
            legendCustomTemplate: 'Error Rate visualization',
          },
        },
      },
```

Reply:
Try kafka as the reference.

There most of signal groups have instance labels equal to broker level.
However, for topic and consumer groups signals instanceLabel is different:

https://github.com/grafana/jsonnet-libs/blob/master/kafka-observ-lib/signals/consumerGroup.libsonnet#L7
instanceLabels: ['topic', 'consumerGroup']
https://github.com/grafana/jsonnet-libs/blob/master/kafka-observ-lib/signals/topic.libsonnet#L7
instanceLabels: ['topic']


Resulting dashboard uses union of these two for variables:
https://github.com/grafana/jsonnet-libs/blob/master/kafka-observ-lib/dashboards.libsonnet#L7-L12
So you can choose different topics, but can't select consumerGroups.